### DISCUSSION TITLE
Is there any way to pass the session configuration when calling the GET admin/engine ?

### DISCUSSION THREAD

**Discussion by HaoYang670** [apache/kyuubi#4804] [General]
Hi. I have an implementation of the `GroupProvider` which users can specify which group they want to be in in the Kyuubi Session configuration.

But when I am trying to send the `GET admin/engine` to check a cluster exists or not, I find there is no way to pass the session configuration, which means I can't specify the group I want to check.

Is there any way to solve this?

**Comment by pan3793** [comment]
cc @turboFei

**Comment by HaoYang670** [comment]
Can we get the Session config from the `fe`, or add a new argument to the function `listEngines` and `deleteEngine`?

**Comment by turboFei** [comment]
```
  def listEngines(
      @QueryParam("type") engineType: String,
      @QueryParam("sharelevel") shareLevel: String,
      @QueryParam("subdomain") subdomain: String,
```

did you set engine type, share level, subdomain params?

**Reply by HaoYang670** [reply]
Yes.

Our custom GroupProvider works like this:

1. Read the group list from Apache Ranger.
2. Read user specified group name from the session config. We define a key `user.group.name`
3. If user doesn't specify the group name, then return the primary group (the head of the group list)
4. otherwise, check `user.group.name` is in the group list or not. If it is, return the group name, otherwise throws exception

**Reply by HaoYang670** [reply]
We implement the GroupProvider like this, so that users can connect to the Kyuubi engine in different groups

**Comment by turboFei** [comment]
I have not used group share level.

In my opinion, I think that you just need to get the correct subdomain from the configuration you said.

**Reply by HaoYang670** [reply]
It is not related to the subdomain. It is related to the group name.

**Comment by turboFei** [comment]
If it is a general requirement, I think it is fine.

**Reply by HaoYang670** [reply]
It is better if we could support this. But this may cause the API change, as we need to let admin GET ENGINE reveive session configs.

**Reply by turboFei** [reply]
would you like to raise a pr? also cc @pan3793 

I think the new parameter should not break the compatibility.

**Reply by HaoYang670** [reply]
Sure, yes I can, as long as you PMCs reach an agreement on whether should we support this.

**Comment by HaoYang670** [comment]
File #4820.
